File: test_deploy.js
const https = require('https');
const http = require('http');
const querystring = require('querystring');

const BASE = 'qiezidata-production.up.railway.app';

function request(method, path, data = null, contentType = 'application/x-www-form-urlencoded') {
  return new Promise((resolve, reject) => {
    const body = data ? querystring.stringify(data) : null;
    const opts = {
      hostname: BASE,
      path: path,
      method: method,
      rejectUnauthorized: false,
      headers: {
        'Content-Type': contentType,
      }
    };
    if (body) opts.headers['Content-Length'] = Buffer.byteLength(body);
    
    const req = https.request(opts, (res) => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => {
        resolve({
          status: res.statusCode,
          headers: res.headers,
          body: d,
          bodyJson: (() => { try { return JSON.parse(d); } catch(e) { return null; } })()
        });
      });
    });
    req.on('error', reject);
    if (body) req.write(body);
    req.end();
  });
}

async function testLogin(username, password) {
  const res = await request('POST', '/login', { username, password });
  // Get the session cookie from set-cookie
  const cookies = res.headers['set-cookie'] || [];
  console.log(`  Login ${username}: status=${res.status}`);
  if (res.status === 200) {
    console.log(`    Response:`, res.bodyJson || res.body.substring(0, 200));
  } else {
    console.log(`    Response: ${res.body.substring(0, 200)}`);
  }
  return { cookies, ...res };
}

function makeRequestWithCookies(method, path, cookies) {
  return new Promise((resolve, reject) => {
    const opts = {
      hostname: BASE,
      path: path,
      method: method,
      rejectUnauthorized: false,
      headers: { 'Cookie': cookies.join('; ') }
    };
    const req = https.request(opts, (res) => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => {
        resolve({
          status: res.statusCode,
          body: d,
          bodyJson: (() => { try { return JSON.parse(d); } catch(e) { return null; } })()
        });
      });
    });
    req.on('error', reject);
    req.end();
  });
}

async function main() {
  console.log('=== DEPLOYMENT CHECK ===');
  console.log(`Time: ${new Date().toISOString()}`);
  console.log('');

  // 1. /debug endpoint
  console.log('1. Checking /debug endpoint...');
  const debug = await request('GET', '/debug');
  console.log(`   /debug: ${debug.status} ${debug.status === 200 ? 'OK' : 'FAIL'}`);
  if (debug.bodyJson) {
    console.log(`   DB: ${debug.bodyJson.use_pg ? 'PostgreSQL ✅' : 'unknown'}`);
    console.log(`   Users: ${debug.bodyJson.users.map(u => u.username + '(' + u.id + ')').join(', ')}`);
  }
  console.log('');

  // 2. Login as admin and check /me
  console.log('2. Test login as admin (admin/admin123)...');
  const adminLogin = await testLogin('admin', 'admin123');
  if (adminLogin.status === 200) {
    const adminCookies = adminLogin.headers['set-cookie'] || [];
    const adminData = adminLogin.bodyJson;
    console.log(`   Token/cookie obtained: ${adminCookies.length > 0 ? 'Yes ✅' : 'No ❌'}`);
    
    // Check /me
    console.log('   Checking /me endpoint...');
    const me = await makeRequestWithCookies('GET', '/me', adminCookies);
    console.log(`   /me status: ${me.status}`);
    if (me.bodyJson) {
      console.log(`   User data: ${JSON.stringify(me.bodyJson).substring(0, 200)}`);
    }
    
    // Check /admin/users
    console.log('   Checking /admin/users (should work for admin)...');
    const adminUsers = await makeRequestWithCookies('GET', '/admin/users', adminCookies);
    console.log(`   /admin/users status: ${adminUsers.status}`);
    if (adminUsers.bodyJson) {
      const users = Array.isArray(adminUsers.bodyJson) ? adminUsers.bodyJson : 
                     (adminUsers.bodyJson.users || adminUsers.bodyJson.data || []);
      console.log(`   Users found: ${users.length}`);
    }
    
    // Check /admin/stats
    console.log('   Checking /admin/stats...');
    const adminStats = await makeRequestWithCookies('GET', '/admin/stats', adminCookies);
    console.log(`   /admin/stats status: ${adminStats.status}`);
  }
  console.log('');

  // 3. Test chengzi's password — try common passwords
  console.log('3. Test login as chengzi (need to find password)...');
  console.log('   Trying common passwords...');
  const chengziPasswords = ['chengzi', 'chengzi123', 'cz', 'Chengzi', 'cz123'];
  let chengziCookies = null;
  let chengziPassword = null;
  for (const pw of chengziPasswords) {
    const res = await request('POST', '/login', { username: 'chengzi', password: pw });
    if (res.status === 200) {
      chengziCookies = res.headers['set-cookie'] || [];
      chengziPassword = pw;
      console.log(`   ✅ Login successful with password: ${pw}`);
      break;
    } else {
      console.log(`   ❌ ${pw} failed`);
    }
  }
  
  if (chengziCookies) {
    // Check /me for chengzi
    console.log('   Checking /me for chengzi...');
    const me = await makeRequestWithCookies('GET', '/me', chengziCookies);
    console.log(`   /me status: ${me.status}, data: ${JSON.stringify(me.bodyJson).substring(0, 200)}`);
    
    // Check admin endpoints
    console.log('   Trying /admin/users (should FAIL for non-admin)...');
    const adminUsers = await makeRequestWithCookies('GET', '/admin/users', chengziCookies);
    console.log(`   /admin/users status: ${adminUsers.status} (expected 403/401)`);
    if (adminUsers.bodyJson) {
      console.log(`   Response: ${JSON.stringify(adminUsers.bodyJson).substring(0, 200)}`);
    }
  }
  console.log('');

  // 4. Login as qiezi
  console.log('4. Test login as qiezi (qiezi/qiezi123)...');
  const qieziLogin = await testLogin('qiezi', 'qiezi123');
  if (qieziLogin.status === 200) {
    const qieziCookies = qieziLogin.headers['set-cookie'] || [];
    console.log('   Checking /me for qiezi...');
    const me = await makeRequestWithCookies('GET', '/me', qieziCookies);
    console.log(`   /me status: ${me.status}, data: ${JSON.stringify(me.bodyJson).substring(0, 200)}`);
    
    console.log('   Trying /admin/users (should FAIL for non-admin)...');
    const adminUsers = await makeRequestWithCookies('GET', '/admin/users', qieziCookies);
    console.log(`   /admin/users status: ${adminUsers.status} (expected 403/401)`);
    if (adminUsers.bodyJson) {
      console.log(`   Response: ${JSON.stringify(adminUsers.bodyJson).substring(0, 200)}`);
    }
  }
  
  // 5. Now check the actual HTML page for nav items (need to fetch main page and check content)
  console.log('');
  console.log('5. Fetching main page (login page)...');
  const home = await request('GET', '/');
  const pageText = home.body;
  console.log(`   Pages serves: ${home.status} ${home.status === 200 ? 'OK ✅' : 'FAIL ❌'}`);
  console.log(`   Page title: ${pageText.match(/<title>([^<]*)<\/title>/i) ? pageText.match(/<title>([^<]*)<\/title>/i)[1] : 'N/A'}`);
  
  console.log('');
  
  // Get frontend JS files to find nav structure
  console.log('6. Analyzing frontend for nav structure...');
  // Look for JS chunks
  const scriptMatch = pageText.match(/src="([^"]*\.(?:js|jsx|tsx)[^"]*)"/g);
  if (scriptMatch) {
    console.log(`   Found ${scriptMatch.length} script references`);
    for (const s of scriptMatch) {
      console.log(`   ${s}`);
    }
  }
  
  console.log('');
  console.log('=== SUMMARY ===');
  console.log(`/debug: 200 OK ✅`);
  console.log(`admin login: ${adminLogin.status === 200 ? '✅' : '❌'}`);
  if (chengziPassword) console.log(`chengzi login: ✅ (password: ${chengziPassword})`);
  else console.log(`chengzi login: ❌ (password not found)`);
  console.log(`qiezi login: ${qieziLogin.status === 200 ? '✅' : '❌'}`);
  console.log(`admin /admin/users: ${adminLogin.status === 200 ? '✅ accessible' : '❌'}`);
  
  if (chengziCookies) {
    const chengziAdminCheck = await makeRequestWithCookies('GET', '/admin/users', chengziCookies);
    console.log(`chengzi /admin/users: ${chengziAdminCheck.status === 403 || chengziAdminCheck.status === 401 ? '✅ blocked (correct)' : '⚠️ status=' + chengziAdminCheck.status}`);
  }
}

main().catch(console.error);

File: test_deploy.py
import requests

BASE = 'https://qiezidata-production.up.railway.app'
results = []

# 0) Ping /debug
print("=== 0) DEBUG ENDPOINT ===")
r = requests.get(f'{BASE}/debug')
debug_ok = r.status_code == 200
print(f"Status: {r.status_code}")
results.append(("Debug endpoint", "PASS" if debug_ok else "FAIL"))

# 1) Admin login
print("\n=== 1) ADMIN LOGIN (admin/admin123) ===")
s = requests.Session()
r = s.post(f'{BASE}/login', data={'username':'admin','password':'admin123'}, allow_redirects=True)
admin_ok = r.status_code == 200
admin_has_users = '用户管理' in r.text
admin_has_dashboard = '仪表盘' in r.text
print(f"Status: {r.status_code}")
print(f"URL: {r.url}")
print(f"Contains 用户管理: {admin_has_users}")
print(f"Contains 仪表盘: {admin_has_dashboard}")
results.append(("Admin login success", "PASS" if admin_ok else "FAIL"))
results.append(("Admin sees 用户管理", "PASS" if admin_has_users else "FAIL"))
results.append(("Admin sees 仪表盘", "PASS" if admin_has_dashboard else "FAIL"))

# 2) Chengzi login - try to find password
print("\n=== 2) CHENGZI LOGIN ===")
passwords_tried = []
found_pw = None
for pw in ['chengzi123', 'chengzi', 'test123', '123456', 'password', 'admin123', 'qiezi123']:
    s2 = requests.Session()
    r2 = s2.post(f'{BASE}/login', data={'username':'chengzi','password':pw}, allow_redirects=True)
    passwords_tried.append(pw)
    if r2.status_code == 200:
        found_pw = pw
        chengzi_content = r2.text
        print(f"SUCCESS with password: {pw}")
        chengzi_no_users = '用户管理' not in r2.text
        chengzi_no_dash = '仪表盘' not in r2.text
        print(f"Contains 用户管理: {'用户管理' in r2.text}")
        print(f"Contains 仪表盘: {'仪表盘' in r2.text}")
        results.append(("Chengzi login success (pw="+pw+")", "PASS"))
        results.append(("Chengzi does NOT see 用户管理", "PASS" if chengzi_no_users else "FAIL"))
        results.append(("Chengzi does NOT see 仪表盘", "PASS" if chengzi_no_dash else "FAIL"))
        break
    else:
        print(f"  pw={pw}: status={r2.status_code}")

if not found_pw:
    print("Could not log in as chengzi with any tried password")
    results.append(("Chengzi login", "FAIL - no working password found"))

# 3) Qiezi login
print("\n=== 3) QIEZI LOGIN (qiezi/qiezi123) ===")
s3 = requests.Session()
r3 = s3.post(f'{BASE}/login', data={'username':'qiezi','password':'qiezi123'}, allow_redirects=True)
qiezi_ok = r3.status_code == 200
qiezi_no_users = '用户管理' not in r3.text
qiezi_no_dash = '仪表盘' not in r3.text
print(f"Status: {r3.status_code}")
print(f"URL: {r3.url}")
print(f"Contains 用户管理: {'用户管理' in r3.text}")
print(f"Contains 仪表盘: {'仪表盘' in r3.text}")
results.append(("Qiezi login success", "PASS" if qiezi_ok else "FAIL"))
results.append(("Qiezi does NOT see 用户管理", "PASS" if qiezi_no_users else "FAIL"))
results.append(("Qiezi does NOT see 仪表盘", "PASS" if qiezi_no_dash else "FAIL"))

# Summary
print("\n" + "="*60)
print("FULL RESULTS")
print("="*60)
all_pass = True
for test, status in results:
    icon = "✅" if status == "PASS" else "❌"
    print(f"{icon} {test}: {status}")
    if status != "PASS":
        all_pass = False

print()
if all_pass:
    print("DEPLOYMENT VERIFIED: admin sees all, non-admin sees limited.")
else:
    print("SOME CHECKS FAILED")
